src/shared/types/ReactPropTypeLocationNames.js

```diff
@@ -16,14 +16,14 @@ import type { ReactPropTypeLocations } from 'ReactPropTypeLocations';
 
 type NamesType = {[key: ReactPropTypeLocations]: string};
 
-var ReactPropTypeLocationNames = {};
+var ReactPropTypeLocationNames: NamesType = {};
```

Comment:
This is the issue that I raised where flow doesn't have a way to have a variable defined only in `__DEV__`, the workaround is to always set the type to what's in `__DEV__` as you just did.